src/Microsoft.OData.Core/SelectedPropertiesNode.cs

```diff
@@ -336,11 +339,62 @@ internal IEnumerable<IEdmNavigationProperty> GetSelectedNavigationProperties(IEd
             return selectedNavigationProperties.Distinct();
         }
 
+        /// <summary>
+        /// Gets all selected structured properties at the current node.
+        /// </summary>
+        /// <param name="structuredType">The current structured type.</param>
+        /// <returns>The selected structured properties at this node level.</returns>
+        /// <remarks>Result does not include the dynamic properties, which can be retrieved by <code>GetSelectedDynamicProperties</code></remarks>
+        internal IEnumerable<IEdmProperty> GetSelectedProperties(IEdmStructuredType structuredType)
+        {
+            if (this.selectionType == SelectionType.Empty)
+            {
+                return EmptyEdmProperties;
+            }
+
+            // We cannot determine the selected properties without the user model. This means we won't be computing the missing properties.
+            if (structuredType == null)
+            {
+                return EmptyEdmProperties;
+            }
+
+            if (this.selectionType == SelectionType.EntireSubtree || this.hasWildcard)
+            {
+                return structuredType.DeclaredProperties;

```

Comment:
>DeclaredProperties [](start = 38, length = 18)

This only returns the properties declared directly on the type; you should use structuredType.Properties to include inherited properties.

Reply:
Why is this by design?  If this is a derived type, why are we only restoring properties declared directly on the derived type and not inherited properties?

---
In reply to: [209788492](https://github.com/OData/odata.net/pull/1106#discussion_r209788492) [](ancestors = 209788492,198059137)

Reply:
Updated to use method structuredType.Properties().

---
In reply to: [213509115](https://github.com/OData/odata.net/pull/1106#discussion_r213509115) [](ancestors = 213509115,209788492,198059137)